app/grandchallenge/workstations/templatetags/workstations.py

```diff
@@ -79,3 +81,25 @@ def workstation_query(  # noqa: C901
         )
 
     return urlencode(query)
+
+
+@register.simple_tag()
+def workstation_session_control_data(
+    *, workstation, context_object, timeout=None, **kwargs
+):
+    create_session_url = reverse(
+        "workstations:workstation-session-create",
+        kwargs={"slug": workstation.slug},
+    )
+    workstation_query_string = get_workstation_query_string(**kwargs)
+    window_identifier = f"workstation-{context_object._meta.app_label}"
+    data_attrs = {
+        "data-create-session-url": create_session_url,
+        "data-workstation-query": workstation_query_string,
+        "data-workstation-window-identifier": window_identifier,
+        "data-timeout": timeout if timeout else False,
```

Comment:
Minor cleanup to make it easier to read

```suggestion
    *, workstation, context_object, timeout=False, **kwargs
):
    create_session_url = reverse(
        "workstations:workstation-session-create",
        kwargs={"slug": workstation.slug},
    )
    workstation_query_string = get_workstation_query_string(**kwargs)
    window_identifier = f"workstation-{context_object._meta.app_label}"
    data_attrs = {
        "data-create-session-url": create_session_url,
        "data-workstation-query": workstation_query_string,
        "data-workstation-window-identifier": window_identifier,
        "data-timeout": timeout,
```